src/RpcServer/RpcServerPlugin.cs

```diff
@@ -54,6 +65,36 @@ protected override void OnSystemLoaded(NeoSystem system)
 
             server.StartRpcServer();
             servers.TryAdd(s.Network, server);
+
+            ApplicationEngine.Log += Ev;
+            Blockchain.Committed += OnCommitted;
+        }
+
+        // It is potentially possible to have dos attack by sending a lot of transactions and logs.
+        // To prevent this, we limit the number of logs to be logged per contract.
+        // If the number of logs is greater than MAX_LOG_EVENTS, we remove the oldest log.
+        private static void Ev(object _, LogEventArgs e)
```

Comment:
All of this log tracking logic should be in RpcServer not RpcServerPlugin.

The separation of RpcServerPlugin and RpcServer is to enable NeoExpress to host RpcServer directly. So any changes to behavior of this plugin must be in RpcServer class so that NeoExpress can use it